argschema/fields/files.py

```diff
@@ -41,23 +60,26 @@ def _validate(self, value):
             path = os.path.dirname(value)
         except Exception as e: # pragma: no cover 
             raise mm.ValidationError("%s cannot be os.path.dirname-ed" % value) # pragma: no cover 
-        try:
-            with tempfile.TemporaryFile(mode='w', dir=path) as tfile:
-                tfile.write('0')
-        except Exception as e:
-            if isinstance(e, OSError):
-                if e.errno == errno.ENOENT:
-                    raise mm.ValidationError(
-                        "%s is not in a directory that exists" % value)
-                elif e.errno == errno.EACCES:
-                    raise mm.ValidationError(
-                        "%s does not appear you can write to path" % value)
-                else:
-                    raise mm.ValidationError(
-                        "Unknown OSError: {}".format(e.message))
-            else:
+        validate_outpath(path)
+
+class OutputDir(mm.fields.Str):
+    """OutputDir is a :class:`marshmallow.fields.Str` subclass which is a path to
+       a location where this module will write files.  Validation will check that 
+       the directory exists and create the directory if it is not present,
+       and will fail validation if the directory cannot be created or cannot be
+       written to.
+    """ 
+
+    def _validate(self,value):
+        if not os.path.isdir(value):
+            try:
+                os.makedirs(value)
```

Comment:
This might not be a huge issue, but I think the race condition introduced by checking for the directory existing can be eliminated by simply adding validation for `errno.EEXIST` in OSError handling for os.makedirs

Reply:
added, good catch